lib/Yield/Calls/Slices.cpp

```diff
@@ -0,0 +1,123 @@
+/// \file Slices.cpp
```

Comment:
What is a slice, exactly? We might find a better name.

Reply:
I'm with you on that.

So far as I'm defining them (in my head) slices are a concatenation of two subtrees (one going forwards and one going backwards) from a selected node in the graph layered so that that node is only mentioned once.

This lets us preserve all the direct and indirect interactions of said node with all the nodes in the graph while also keeping it as slim as possible by omitting all the unrelated nodes.

Reply:
But is this a generic concept or is it a *call graph* slice? If it's only for call graphs, I'd call it `CallGraphSlice`.

Reply:
Technically it's generic, but I'm not sure there's much benefit in that.

Nothing prevents you from, say, getting a slice of a `ControlFlowGraph` and stuff, but why would you want it?

The main limiting factor is that it works on `yield::Graph`s.

Reply:
Considering that `Slices.h` only works with call graphs concepts, let's rename this to `CallGraphSlices`.
We can generalize names in the future, if needed. But I don't think we'll have to. Let's be pragmatic.

Reply:
Sounds good.

Reply:
Now that I got to renaming these, it feels really weird to name them `Yield/Calls/CallGraphSlices`, it feels like the data gets duplicated. Should I still go with it, or is the directory name (which should probably become `CallGraphs` for consistency) good enough?

Reply:
* +1 for `s/Calls/CallGraph/`
* The name should reflect the class name. So `CallGraphSlices` works better, in absence of namespaces.

Reply:
Done: https://github.com/revng/revng/pull/191/commits/c0dafcbd3726174661b49e13ff89ae5cf03e91ed.